x/proof/module/query_claim_test.go

```diff
@@ -223,11 +236,11 @@ func TestClaim_List(t *testing.T) {
 		var resp types.QueryAllClaimsResponse
 		require.NoError(t, net.Config.Codec.UnmarshalJSON(out.Bytes(), &resp))
 
+		require.Equal(t, numSuppliers, int(resp.Pagination.Total))
 		require.ElementsMatch(t,
 			nullify.Fill(expectedClaims),
 			nullify.Fill(resp.Claims),
 		)
-		require.Equal(t, numSuppliers, int(resp.Pagination.Total))
```

Comment:
What was the rationale behind swapping the order of these? 

**TL;DR, I think the result here is that the `ElementsMatch()` assertion error won't be printed, leaving you wondering why the total number assertion failed.**

IIRC, doing this makes it difficult to determine why the expected total number doesn't match the actual total number; whereas if the `ElementsMatch()` assertion is first, its error will contain the answer. The issue is that, since we're using `require` (as opposed to `assert`), the test fails right away on whichever fails first. Since both assertions would fail if the total number assertion fails, this reduces to whichever **is** first (in the code). 

I think this concern/logic applies anywhere that we make a similar pair of correlated assertions (e.g. elements & total).

(Same above.)

Reply:
I'm going to give you the perspective of a real developer (me) who broke a real test (this) while working on this feature.

1. I saw A LOT of output and my immediate thought was: "wtf is going on".
2. I went into the test and saw where the assertion happened. 
3. I saw that before I check "what the content is", I wanted to do the simple check of "is the content of the right length" because I was hitting zeroes.

**Conclusion**: Asserting on something simple and easy to interpret FIRST makes life easier from my anecdotal experience.